### PR TITLE
[risk=low][RW-3890] Make display name and terra name distinct in API unit tests

### PR DESCRIPTION
**Pull request by jmthibault79**
Part of the "Terra Name" project - distinguish between Display Name and Terra Name in API unit tests, where we often conflated the two.

This change is strictly test code, so I did not run locally.

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [ ] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [x] None of the above apply to this change.

### CODE REVIEW

File: api/src/test/java/org/pmiops/workbench/utils/TestMockFactory.java [hunk 1]
```diff
@@ -111,10 +110,8 @@ public class TestMockFactory {
               .setName(DbAccessModuleName.PUBLICATION_CONFIRMATION)
               .setExpirable(true));
 
-  // TODO there's something off about how "workspaceName" here works.  Investigate.
-  // For best results, use a lowercase-only workspaceName.
-  // To me, this hints at a firecloudName/aouName discrepancy somewhere in here.
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/utils/TestMockFactory.java:116]: yep

File: api/src/test/java/org/pmiops/workbench/utils/TestMockFactory.java [hunk 2]
```diff
@@ -163,13 +161,13 @@ public static Workspace createWorkspace(String workspaceNameSpace, String worksp
...
+  public static RawlsWorkspaceDetails createTerraWorkspace(
+      String namespace, String terraName, String creator) {
     return new RawlsWorkspaceDetails()
-        .namespace(ns)
-        .workspaceId(ns)
-        .name(name)
+        .namespace(namespace)
+        .name(terraName)
         .createdBy(creator)
+        .workspaceId(WORKSPACE_TERRA_UUID)
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/utils/TestMockFactory.java:170]: this is not the workspace name, terraName, or namespace.

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java [hunk 3]
```diff
@@ -298,7 +306,7 @@ private DbWorkspace addMockedWorkspace(
   private DbWorkspace addMockedWorkspace(DbWorkspace dbWorkspace) {
 
     mockRawlsWorkspaceResponse(
-        dbWorkspace.getWorkspaceId(),
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java:301]: this was the wrong one

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java [hunk 4]
```diff
@@ -335,20 +343,6 @@ private DbWorkspace addMockedPublishedWorkspace(
     return dbWorkspace;
   }
 
-  private RawlsWorkspaceResponse mockRawlsWorkspaceResponse(
-      long workspaceId,
```
Comment (jmthibault79) [inline api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java:339]: this version was not valid

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java [hunk 5]
```diff
@@ -219,31 +219,35 @@ public void setUp() {
...
-            .name(workspaceName)
+            .workspaceId(workspaceTerraUuid)
+            .name(workspaceTerraName)
             .namespace(workspaceNamespace);
 
     RawlsWorkspaceResponse mockWorkspaceResponse = new RawlsWorkspaceResponse();
     mockWorkspaceResponse.workspace(mockWorkspace);
     mockWorkspaceResponse.accessLevel(accessLevel);
+
+    doReturn(mockWorkspaceResponse)
```
Comment (evrii) [inline api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java:236]: Was this not needed before?

**Review by evrii — APPROVED** [review]
Seems reasonable to me

File: api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java (hunk 5 above)
Comment (jmthibault79) [reply api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java:236]: There were two `mockRawlsWorkspaceResponse()` methods previously, and this is the result of merging them.  The one github chose as the diff didn't have it, but the other one did.

Considering the name of the method, it's strange that one of them didn't actually mock the rawls workspace response!

**Pull request merged**